_drafts/release-workflows.md

```diff
@@ -0,0 +1,383 @@
+- Pre-releases
+
+## Proposed solution
+
+Support the following type of branches:
+- [Single release branch](#single-release-branch)
+- [Non release branch](#non-release-branch)
+- [Future branches](#future-branches)
+- [LTS branches](#lts-branches)
+- [Pre-releases](#pre-releases)
+
+## Detailed design
+
+**Terminology:**
+
+| Term                | Definition                                                                                                                                                                         |
+|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Default branch      | Git branch that contains the stable code (made available to all users). Usually named `master`.                                                                                    |
+| Future branches     | Permanent Git branch that contains the unstable code (made available only to early adopters). Usually named `next`.                                                                |
+| Channel             | Distribution channels used to limit the accessibility of a release to a subset of users. For example `dist-tags` in `npm`.                                                         |
+| Default channel     | Distribution channel to make stable version available to every users. Usually named `latest`.                                                                                      |
+| Future channels     | Distribution channels to make unstable version available to early adopters. Usually named `next`.                                                                    |
+| Pre-release version | version that doesn't follow the regular `semver` increment pattern. For example any increment of `2.0.0-beta.1` is `2.0.0-beta.2`, then `2.0.0-beta.2` etc...                      |
+| Version             | version following the `semver` rules. For example, a `minor` increment of `1.0.0` is `1.1.0`.                                                                                      |
+| Module              | A software meant to be used as a dependency of other software. It's not meant to be used by final users.                                                                           |
+| Application         | A software meant to be used by final users. No other software depends on it.                                                                                                    |
+| Module version      | The version of a module which is meant for machines, not humans. For example a `major` release indicates a breaking change but doesn't cary any marketing or emotional value. |
+| Application version | The version of an application which is meant for humans. For example a `major` release has a marketing meaning.                                                              |
```

Comment:
> do we need the definitions of module version and application version in here?

Not really. It should be self-explanatory when it's used later on. I can remove it.

> can we replace minor & major with breaking and feature for clarity?

Not if replacing would be more clear. It might be the best terms but they are the one users by the SemVer specs. I'll make a change to mention both like: `breaking release (major in SemVer terminology)`.

Reply:
yeah I would avoid major/minor as much as possible. Maybe add Breaking/Feature/Fix versions to the terminology once and mention there the counterparts in the SemVer specification?

Reply:
maybe it would be worth linking to external sources for each?
* [semver spec](https://semver.org/) for `major` & `minor`
* [greenkeeper blog article](https://blog.greenkeeper.io/introduction-to-semver-d272990c44f2) for `breaking` & `feature`